source/common/config/version_converter.cc

```diff
@@ -1,12 +1,64 @@
 #include "common/config/version_converter.h"
 
+#include "common/common/assert.h"
 #include "common/config/api_type_oracle.h"
 
+#include "absl/strings/match.h"
+
 namespace Envoy {
 namespace Config {
 
 namespace {
 
+// TODO(htuch): refactor these message visitor patterns into utility.cc and share with
+// MessageUtil::checkForUnexpectedFields.
```

Comment:
I had a similar comment also about the redacting stuff that @mergeconflict is working on. Can the utility be used there also if possible?

Reply:
Probably. I need to do use this visitor in my next PR, so I reckon we can merge this as is, and I will move this to a more general location there